### PR TITLE
Fix `selector-attribute-name-disallowed-list` reported ranges

### PR DESCRIPTION
**Pull request by ryo-manba**
> Which issue, if any, is this issue related to?

See https://github.com/stylelint/stylelint/issues/7904

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/rules/selector-attribute-name-disallowed-list/index.mjs [hunk 1]
```diff
@@ -37,24 +37,26 @@ const rule = (primary) => {
 				return;
 			}
 
-			parseSelector(ruleNode.selector, result, ruleNode)?.walkAttributes((attributeNode) => {
-				const attributeName = attributeNode.qualifiedAttribute;
+			flattenNestedSelectorsForRule(ruleNode, result).forEach(({ selector }) => {
```
Comment (romainmenke) [inline lib/rules/selector-attribute-name-disallowed-list/index.mjs:40]: Do we need to switch to `flattenNestedSelectorsForRule`?

This rule only looks at individual attribute selector parts and doesn't do any analysis of the entire selector. So resolving nested selectors shouldn't be needed.

I think we can use the `getRuleSelector` utility instead.

e.g.

```js
parseSelector(getRuleSelector(ruleNode), result, ruleNode)?.walkAttributes((attributeNode) => {
```

**Review by romainmenke — COMMENTED** [review]
Thank you for working on this issue @ryo-manba 🙇 
I've left a question about a potentially smaller change.

File: lib/rules/selector-attribute-name-disallowed-list/index.mjs (hunk 1 above)
Comment (ryo-manba) [reply lib/rules/selector-attribute-name-disallowed-list/index.mjs]: Got it, using `getRuleSelector` helps to include comments in the selector. I've made the update.

[refactor to use getRuleSelector](https://github.com/stylelint/stylelint/pull/8037/commits/4d81b4ddea71f73da346ead5115e9d2a7db70b8d)

**Review by romainmenke — APPROVED** [review]
LGTM!
Thank you for your contribution.

(leaving this open in case other maintainers want to take a look)

File: lib/rules/selector-attribute-name-disallowed-list/__tests__/index.mjs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import rule from '../index.mjs';
+import { stripIndent } from 'common-tags';
 const { messages, ruleName } = rule;
```
Comment (romainmenke) [inline lib/rules/selector-attribute-name-disallowed-list/__tests__/index.mjs:3]:
```suggestion
import { stripIndent } from 'common-tags';

import rule from '../index.mjs';
const { messages, ruleName } = rule;
```

@ryo-manba We split 3rd party and internal module imports.
3rd party modules come first and are separated by an empty line before listing the internal module imports.

**Pull request merged**